pkg/restapi/edv/operation/operations.go

```diff
@@ -119,15 +122,25 @@ func (c *Operation) createDocumentHandler(rw http.ResponseWriter, req *http.Requ
 
 		_, err = rw.Write([]byte(err.Error()))
 		if err != nil {
-			log.Errorf("Failed to write response for document creation failure: %s",
-				err.Error())
+			log.Errorf("Failed to write response for document creation failure: %s", err.Error())
 		}
 
 		return
 	}
 
 	vars := mux.Vars(req)
-	vaultID := vars[vaultIDPathVariable]
+
+	vaultID, err := url.PathUnescape(vars[vaultIDPathVariable])
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+
+		_, err = rw.Write([]byte(fmt.Sprintf("unable to escape vault ID path variable: %s", err.Error())))
+		if err != nil {
+			log.Errorf("Failed to write response for vault ID unescaping failure: %s", err.Error())
+		}
+
+		return
+	}
```

Comment:
I see similar block is repeated many times in this file, follow [this](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/restapi/operation/common/operation.go#L163) pattern

Reply:
done